frontend/lib/wallets/Coinbase.ts

```diff
@@ -0,0 +1,81 @@
+import CoinbaseWalletSDK, { CoinbaseWalletProvider } from '@coinbase/wallet-sdk';
+import { ethers } from 'ethers';
+import { getChainIdByNetworkType, getNetworkTypeByChainId } from 'auctions-core/src/network';
+import { setSigner } from 'auctions-core/src/signer';
+import { formatToHexWithoutPad } from 'auctions-core/helpers/format';
+import CoinbaseLogo from '~/assets/icons/wallets/coinbase.svg';
+import AbstractWallet from '~/lib/wallets/AbstractWallet';
+
+export default class Coinbase extends AbstractWallet {
+    public static title = 'Coinbase Wallet';
+    public static icon = CoinbaseLogo;
+    public static downloadUrl = 'https://www.coinbase.com/wallet/downloads';
+
+    private static provider?: CoinbaseWalletProvider;
+
+    private addresses?: string[];
+
+    public get address() {
+        if (Array.isArray(this.addresses) && this.addresses[0]) {
+            return this.addresses[0].toLowerCase();
+        }
+    }
+
+    getProvider(): ethers.providers.JsonRpcProvider {
+        const rpcUrl = process.env.RPC_URL;
```

Comment:
If you take RPC_URL directly from the env var, how does the network switch suppose to work? `RPC_URL` is different per network and as I several times pointed out, you should instead use
```ts
const rpcUrl = getNetworks()[netoworkName].url
```